tools/autograd/templates/Functions.h

```diff
@@ -35,7 +35,8 @@ inline c10::List<c10::optional<Tensor>> unpack_opt_list(at::ArrayRef<SavedVariab
   torch::List<c10::optional<Tensor>> result;
   result.reserve(xs.size());
   for (const SavedVariable& v : xs) {
-    result.push_back(v.unpack());
+    auto var = v.unpack();
```

Comment:
Why do you need to change this?

Reply:
**The problem**: in `IListIterator::operator*` I try to convert an `IValue` (boxed element) into an `OptionalTensorRef`. There, I'm mapping:

- `ivalue.isNone()` => `OptionalTensorRef{}`
- otherwise => implicitly conversion from `ivalue.toTensor()`

However, I can't create an `OptionalTensorRef` from an undefined tensor (there's a check for that).

------

While this change solve the issue above, I could, instead, change the mapping I do in `IList`:

- `ivalue.isNone() && !ivalue.toTensor().defined()` => `OptionalTensorRef{}`
- otherwise => implicitly conversion from `ivalue.toTensor()`

Do you think that's a better approach?

Reply:
For your use case I am not sure. But this change does change the semantic of the current function.
I am not familiar about all the call site for this function but we should make sure that this change in semantic is ok there.

Reply:
You are right. I think it's best to handle this inside `IList`.

Reply:
This still needs updating right?